src/screens/Signup/index.tsx

```diff
@@ -0,0 +1,92 @@
+import { useTranslation } from 'react-i18next';
+import { useForm } from 'react-hook-form';
+
+import Button from 'components/Button';
+import Input from 'components/Input';
+
+import logo from './assets/wolox-logo.png';
+import styles from './styles.module.scss';
+
+interface SignupForm {
+  firstName: string;
+  lastName: string;
+  email: string;
+  password: string;
+  passwordConfirmation: string;
+}
+
+function Signup() {
+  const { t, i18n } = useTranslation();
+  const {
+    register,
+    handleSubmit,
+    watch,
+    formState: { errors }
+  } = useForm<SignupForm>();
+
+  const password = watch('password');
+
+  const onSubmit = handleSubmit((values) => console.log({ ...values, locale: 'en' }));
+
+  const handleChangeLanguage = () => {
+    i18n.changeLanguage(i18n.language === 'es' ? 'en' : 'es');
+  };
+
+  return (
+    <div className={styles.loginWrapper}>
+      <img src={logo} alt="Wolox Logo" className={styles.img} />
+      <form action="" className={`column ${styles.form}`} onSubmit={onSubmit}>
+        <Input
+          inputType="text"
+          name="firstName"
+          label={t('Signup:firstName')}
+          inputRef={register({ required: `${t('Signup:requiredInput')}` })}
```

Comment:
Why not?
```suggestion
          inputRef={register({ required: t('Signup:requiredInput') })}
```

Reply:
It gives me 'No overload matches this call.' error and I'm not sure why 🥲

Reply:
Mmm that's because there's is no match (overlap) between what is expected as a parameter and what is being passed as a param.
For more info about what that error message could mean [check this out](https://lwkchan.com/blog/2020-09-20-typescript-errors-for-humans-no-overload-matches-this-call/)